packages/components/src/InputText/InputText.tsx

```diff
@@ -52,12 +61,53 @@ function InputTextInternal(
 
   return (
     <FormField
+      {...props}
       type={props.multiline ? "textarea" : "text"}
       ref={inputRef}
-      {...props}
+      onChange={handleChange}
+      rows={rowRange.min}
     />
   );
 
+  function handleChange(newValue: string) {
+    props.onChange && props.onChange(newValue);
+
+    if (inputRef && inputRef.current instanceof HTMLTextAreaElement) {
+      resize(inputRef.current);
+    }
+  }
+
+  function getRowRange(): RowRange {
+    if (props.rows === undefined) {
+      return { min: 3, max: 3 };
+    } else if (typeof props.rows === "object") {
+      return { min: props.rows.min, max: props.rows.max };
+    } else {
+      return { min: props.rows, max: props.rows };
+    }
+  }
+
+  function resize(textArea: HTMLTextAreaElement) {
+    if (rowRange.min === rowRange.max) return;
+
+    textArea.style.height = "auto";
+    textArea.style.height = textAreaHeight(textArea) + "px";
+  }
+
+  function textAreaHeight(textArea: HTMLTextAreaElement) {
+    const maxHeight = rowRange.max * getLineHeight(textArea);
+    return Math.min(textArea.scrollHeight, maxHeight);
+  }
+
+  function getLineHeight(textArea: HTMLTextAreaElement) {
+    const { fontSize, lineHeight } = window.getComputedStyle(textArea);
+    const lh =
+      lineHeight === "normal"
+        ? parseFloat(fontSize) * 1.2
```

Comment:
@eddysims

I got this from Eddy's example, where does the magic `1.2` number come from? Should this be a dynamically retrieved value from somewhere?

Reply:
yeah, this isnt the best way... we can do something like

```
const lineHeight = window
        .getComputedStyle(inputRef.current)
        .getPropertyValue("line-height");
```

*note* I grabbed this from a slack message, so it might not be exactly correct, but it should be similar to what you want.

Reply:
Ah it's the line height. Makes sense now 👍